src/controllers/legacycontrollersettings.h

```diff
@@ -127,9 +126,56 @@ class AbstractLegacyControllerSetting : public QObject {
     QString m_description;
 };
 
+/// @brief Base setting class generic
+/// @tparam T The type of value this setting type is holding
+template<class T>
+class LegacyControllerSettingMixin : public AbstractLegacyControllerSetting {
+  public:
+    bool isDefault() const override {
+        return m_savedValue == m_defaultValue;
+    }
+    bool isDirty() const override {
+        return m_savedValue != m_editedValue;
+    }
+
+    void save() override {
+        m_savedValue = m_editedValue;
+    }
+
+    void reset() override {
+        m_editedValue = m_defaultValue;
+        emit valueReset();
+    }
+
+  protected:
+    LegacyControllerSettingMixin(const QDomElement& element)
```

Comment:
What mm means "Mixin" ?

Reply:
https://en.m.wikipedia.org/wiki/Mixin

I must confess I didn't know it either a few seconds ago :smiley: